docs/data/charts/scatter/ScatterCustomShape.tsx

```diff
@@ -0,0 +1,205 @@
+import * as React from 'react';
+import { ScatterChart, ScatterMarkerProps } from '@mui/x-charts/ScatterChart';
+
+const data = [
+  {
+    id: 'data-0',
+    x1: 329.39,
+    y1: 443.28,
+    y2: 153.9,
+  },
+  {
+    id: 'data-1',
+    x1: 96.94,
+    y1: 110.5,
+    y2: 217.8,
+  },
+  {
+    id: 'data-2',
+    x1: 336.35,
+    y1: 175.23,
+    y2: 286.32,
+  },
+  {
+    id: 'data-3',
+    x1: 159.44,
+    y1: 195.97,
+    y2: 325.12,
+  },
+  {
+    id: 'data-4',
+    x1: 188.86,
+    y1: 351.77,
+    y2: 144.58,
+  },
+  {
+    id: 'data-5',
+    x1: 143.86,
+    y1: 43.253,
+    y2: 146.51,
+  },
+  {
+    id: 'data-6',
+    x1: 202.02,
+    y1: 376.34,
+    y2: 309.69,
+  },
+  {
+    id: 'data-7',
+    x1: 384.41,
+    y1: 31.514,
+    y2: 236.38,
+  },
+  {
+    id: 'data-8',
+    x1: 256.76,
+    y1: 231.31,
+    y2: 440.72,
+  },
+  {
+    id: 'data-9',
+    x1: 143.79,
+    y1: 108.04,
+    y2: 20.29,
+  },
+  {
+    id: 'data-10',
+    x1: 103.48,
+    y1: 321.77,
+    y2: 484.17,
+  },
+  {
+    id: 'data-11',
+    x1: 272.39,
+    y1: 120.18,
+    y2: 54.962,
+  },
+  {
+    id: 'data-12',
+    x1: 23.57,
+    y1: 366.2,
+    y2: 418.5,
+  },
+  {
+    id: 'data-13',
+    x1: 219.73,
+    y1: 451.45,
+    y2: 181.32,
+  },
+  {
+    id: 'data-14',
+    x1: 54.99,
+    y1: 294.8,
+    y2: 440.9,
+  },
+  {
+    id: 'data-15',
+    x1: 134.13,
+    y1: 121.83,
+    y2: 273.52,
+  },
+  {
+    id: 'data-16',
+    x1: 12.7,
+    y1: 287.7,
+    y2: 346.7,
+  },
+  {
+    id: 'data-17',
+    x1: 176.51,
+    y1: 134.06,
+    y2: 74.528,
+  },
+  {
+    id: 'data-18',
+    x1: 65.05,
+    y1: 104.5,
+    y2: 150.9,
+  },
+  {
+    id: 'data-19',
+    x1: 162.25,
+    y1: 413.07,
+    y2: 26.483,
+  },
+  {
+    id: 'data-20',
+    x1: 68.88,
+    y1: 74.68,
+    y2: 333.2,
+  },
+  {
+    id: 'data-21',
+    x1: 95.29,
+    y1: 360.6,
+    y2: 422.0,
+  },
+  {
+    id: 'data-22',
+    x1: 390.62,
+    y1: 330.72,
+    y2: 488.06,
+  },
+];
+
+export default function ScatterCustomShape() {
+  return (
+    <ScatterChart
+      width={600}
+      height={300}
+      series={[
+        {
+          id: '1',
+          label: 'Series A',
+          data: data.map((v) => ({ x: v.x1, y: v.y1, id: v.id })),
+          markerSize: 6,
+          labelMarkType: 'square',
+        },
+        {
+          id: '2',
+          label: 'Series B',
+          data: data.map((v) => ({ x: v.x1, y: v.y2, id: v.id })),
+          markerSize: 6,
+          labelMarkType: 'line',
+        },
+      ]}
+      slots={{ marker: CustomMarker }}
+    />
+  );
+}
+
+function CustomMarker({
+  series,
+  x,
+  y,
+  isHighlighted,
+  isFaded,
+  dataIndex,
+  color,
+  ownerState: _,
```

Comment:
`useSlotProps` requires an `ownerState` prop, so we need to destructure it to avoid being spread into the object. 

Also, not using `ownerState` means that we can cause a breaking change by adding more props. I wonder if we should move to `ownerState` all props that can't/shouldn't be directly spread onto the element.

Reply:
You could just not pass it in 😆

As I said, the `useSlotProps` is not very good for our needs, I'll take a look if we can use the `consumeSlots` in this case

Reply:
I've pushed a commit that uses the `consumeSlots`, it should make the api simpler, and we get rid of an intermediate component 😅

Not sure about the perf impacts though

https://github.com/mui/mui-x/pull/16640/commits/b67ce0960752ae125814b80ce60d4367076f521e

Reply:
> You could just not pass it in 😆

There's a TypeScript error if I do that 😄 

> I've pushed a commit 

Thanks! I'll see if I can improve those perf benchmarks.